File: courses/machine_learning/Templates/Python/regression_template.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Import data
dataset = pd.read_csv("data.csv")

# Extracting features (X) and target variable (y)
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values

# Fitting the regression model to the dataset
regressor = # Regression model
regressor.fit(X, y)

# Predicting new result with regression model
# Replace 'new_data' with the data you want to predict
new_data = np.array([[feature_value1, feature_value2]])  # Example
y_pred = regressor.predict(new_data)

# Visualizing regression model
plt.scatter(X, y, color="red")
plt.plot(X, regressor.predict(X), color="blue")
plt.title("Regression Model")
plt.xlabel("X label")
plt.ylabel("Y label")
plt.show()

File: courses/machine_learning/Templates/R/regression_template.R
# Regression template

# Import data
dataset = read.csv("data.csv")
# dataset = dataset[2:3]

# Fitting the regression model to the dataset
# create regressor

# Predicting new result whit regression model
y_pred = predict(regressor, newdata = data.frame(feature = predict))

library(ggplot2)
# Visualising regression model
ggplot()+
  geom_point(aes(x = dataset$X, y = dataset$Y),
             colour = "red")+
  geom_line(aes(x = dataset$Level, y = predict(regressor,newdata = dataset)),
                colour = "blue") +
  ggtitle("Regression regression model") +
  xlab("X label") +
  ylab("Y label")


